lib/matplotlib/figure.py

```diff
@@ -913,6 +913,42 @@ def _break_share_link(ax, grouper):
         # Break link between any twinned axes
         _break_share_link(ax, ax._twinned_axes)
 
+    def clf(self, keep_observers=False):
+        """
+        Clear the figure.
+        Set *keep_observers* to True if, for example,
+        a gui widget is tracking the Axes in the figure.
+        """
+        self.suppressComposite = None
+        self.callbacks = cbook.CallbackRegistry()
+
+        # first clear the axes in any subfigures
+        for subfig in self.subfigs:
+            subfig.clf(keep_observers=keep_observers)
+        self.subfigs = []
+
+        for ax in tuple(self.axes):  # Iterate over the copy.
+            ax.cla()
+            self.delaxes(ax)  # Remove ax from self._axstack.
+        self._axstack = _AxesStack()
```

Comment:
I don't think this is correct, as `Figure` gets its own `_AxesStack`, but `SubFigure` uses its parent's, and replacing it here would break that link.